cranelift/codegen/meta/src/shared/instructions.rs

```diff
@@ -2174,6 +2174,37 @@ pub(crate) fn define(
         .operands_out(vec![a, c_if_out]),
     );
 
+    {
+        let code = &Operand::new("code", &imm.trapcode);
+
+        let i32_64 = &TypeVar::new(
+            "i32_64",
+            "A 32 or 64-bit scalar integer type",
+            TypeSetBuilder::new().ints(32..64).build(),
+        );
+
+        let a = &Operand::new("a", i32_64);
+        let x = &Operand::new("x", i32_64);
+        let y = &Operand::new("y", i32_64);
+        ig.push(
+            Inst::new(
+                "iadd_overflow_trap",
+                r#"
+            Add 32 or 64-bit integers, and trap if overflow occurs.
+
+            This is the same as `iadd_cout` but traps instead of returning a carry flag.
```

Comment:
I think we can just say

> This is the same as `iadd` but traps on overflow instead of wrapping.

since `iadd` is a slightly less esoteric instruction than `iadd_cout`.

Reply:
That's a good point, thanks!